Switch ARKit from soft linking to weak linking

achan00 opened the pull request:
#### de81e985f47bec8708cade4a5f7dfafd696d3c72
<pre>
Switch ARKit from soft linking to weak linking
<a href="https://bugs.webkit.org/show_bug.cgi?id=254218">https://bugs.webkit.org/show_bug.cgi?id=254218</a>

Reviewed by Tim Horton and David Kilzer.

- Update linker flags to weak link ARKit
- Remove soft linking files for ARKit and stop building/importing them.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Shared/Cocoa/ARKitSoftLink.h: Removed.
* Source/WebKit/Shared/Cocoa/ARKitSoftLink.mm: Removed.
* Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/261980@main">https://commits.webkit.org/261980@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/897a09dfd4ce19ae93637f8425f92ade4a146e12

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/25 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21 "3 api tests failed or timed out") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15 "48 flakes 178 failures 1 missing results") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/22 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/15 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/41 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/22 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer reviewed (approved):
r=me

I'll state the obvious here, but we should wait until iOS builds+tests (at minimum) pass before committing this.

ddkilzer commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -61,6 +61,14 @@ WK_APPLE_PUSH_SERVICE_LDFLAGS_IOS_SINCE_16 = -framework ApplePushService;
+WK_ARKIT_LDFLAGS_iphonesimulator = -weak_framework ARKit;
+WK_ARKIT_LDFLAGS_maccatalyst = -weak_framework ARKit;
+WK_ARKIT_LDFLAGS_macosx = $(WK_ARKIT_LDFLAGS_macosx$(WK_MACOS_1200));
+WK_ARKIT_LDFLAGS_macosx_MACOS_SINCE_1200 = -weak_framework ARKit;
```
Deleting these two lines should fix the macOS build.

achan00 committed:
Switch ARKit from soft linking to weak linking
https://bugs.webkit.org/show_bug.cgi?id=254218

Reviewed by Tim Horton and David Kilzer.

- Update linker flags to weak link ARKit
- Remove soft linking files for ARKit and stop building/importing them.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Shared/Cocoa/ARKitSoftLink.h: Removed.
* Source/WebKit/Shared/Cocoa/ARKitSoftLink.mm: Removed.
* Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/261980@main
